mockstore: panic happens for test TestMvccTxnRead

## Bug Report

### 1. Minimal reproduce step (Required)

Test `TestMvccTxnRead` failed with following stack:

```
[2021-12-06T09:50:51.934Z] --- FAIL: TestMvccTxnRead (1.51s)
[2021-12-06T09:50:51.934Z] panic: runtime error: invalid memory address or nil pointer dereference [recovered]
[2021-12-06T09:50:51.934Z] 	panic: runtime error: invalid memory address or nil pointer dereference
[2021-12-06T09:50:51.934Z] [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1490ffc]
[2021-12-06T09:50:51.934Z] 
[2021-12-06T09:50:51.934Z] goroutine 116 [running]:
[2021-12-06T09:50:51.934Z] testing.tRunner.func1.2(0x15d0640, 0x2bf3be0)
[2021-12-06T09:50:51.934Z] 	/usr/local/go/src/testing/testing.go:1143 +0x332
[2021-12-06T09:50:51.934Z] testing.tRunner.func1(0xc0005ebe00)
[2021-12-06T09:50:51.934Z] 	/usr/local/go/src/testing/testing.go:1146 +0x4b6
[2021-12-06T09:50:51.934Z] panic(0x15d0640, 0x2bf3be0)
[2021-12-06T09:50:51.934Z] 	/usr/local/go/src/runtime/panic.go:965 +0x1b9
[2021-12-06T09:50:51.934Z] github.com/pingcap/tidb/store/mockstore/unistore/tikv.(*regionCtx).AcquireLatches(0xc0004d1a40, 0xc000922068, 0x1, 0x1)
[2021-12-06T09:50:51.934Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/store/mockstore/unistore/tikv/region.go:251 +0x1bc
[2021-12-06T09:50:51.934Z] github.com/pingcap/tidb/store/mockstore/unistore/tikv.(*MVCCStore).Commit(0xc010166080, 0xc129192500, 0xc00fd22198, 0x1, 0x1, 0x5, 0xa, 0x0, 0x0)
[2021-12-06T09:50:51.934Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/store/mockstore/unistore/tikv/mvcc.go:913 +0x219
[2021-12-06T09:50:51.934Z] github.com/pingcap/tidb/store/mockstore/unistore/tikv.MustCommit(0xc036f1e1b7, 0x3, 0x3, 0x5, 0xa, 0xc0008526c0)
[2021-12-06T09:50:51.934Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/store/mockstore/unistore/tikv/mvcc_test.go:390 +0x12f
[2021-12-06T09:50:51.934Z] github.com/pingcap/tidb/store/mockstore/unistore/tikv.TestMvccTxnRead(0xc0005ebe00)
[2021-12-06T09:50:51.934Z] 	/home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/store/mockstore/unistore/tikv/mvcc_test.go:967 +0x5da
[2021-12-06T09:50:51.934Z] testing.tRunner(0xc0005ebe00, 0x187ec60)
[2021-12-06T09:50:51.934Z] 	/usr/local/go/src/testing/testing.go:1193 +0xef
[2021-12-06T09:50:51.934Z] created by testing.(*T).Run
[2021-12-06T09:50:51.934Z] 	/usr/local/go/src/testing/testing.go:1238 +0x2b3
```

From https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/50372/pipeline, PR https://github.com/pingcap/tidb/pull/30306. I don't think it is related to #30306.

### 4. What is your TiDB version? (Required)

It happens in the CI of the `master` branch.

--- a/store/mockstore/unistore/tikv/region.go
+++ b/store/mockstore/unistore/tikv/region.go
@@ -248,7 +248,13 @@ func (ri *regionCtx) AcquireLatches(hashVals []uint64) {
 	dur := time.Since(start)
 	metrics.LatchWait.Observe(dur.Seconds())
 	if dur > time.Millisecond*50 {
-		log.S().Warnf("region %d acquire %d locks takes %v, waitCnt %d", ri.meta.Id, len(hashVals), dur, waitCnt)
+		var id string
+		if ri.meta == nil {
+			id = "unknown"
+		} else {
+			id = strconv.FormatUint(ri.meta.Id, 10)
+		}
+		log.S().Warnf("region %s acquire %d locks takes %v, waitCnt %d", id, len(hashVals), dur, waitCnt)
 	}
 }
 

